### PR TITLE
Prototype: Quick, dirty implementation of ad lightboxes

### PR DESCRIPTION
**Pull request by alanorozco**
**This is not production ready.**

This PR corresponds to an in-progress prototype. Using an `amp-lightbox` in an ad will not pass validation and will not function properly under normal circumstances.

Guarded under `amp-lightbox-a4a-proto` experiment.

/cc @jasti 

### Future work

- Move iframe resizing logic out of the `amp-lightbox` module.
- Implement `amp-ad-banner` extension (design pending).
- Test fully for scrolling position/use viewport service.
- **Unit tests.** The current lightbox implementation does not include any automated tests.

### CODE REVIEW

File: extensions/amp-lightbox/0.1/amp-lightbox.js [hunk 1]
```diff
@@ -19,14 +19,94 @@ import {Gestures} from '../../../src/gesture';
...
+ * @param {!Window} topLevelWindow
+ */
+function enterFrameFullOverlayMode(iframe, topLevelWindow) {
+  // TODO(alanorozco): use viewport service
+  // TODO(alanorozco): move ad banner resizing logic to its extension class
+  // TODO(alanorozco): check for FriendlyIframeEmbed.win.document as iframeDoc
+  //                   fallback.
+  const iframeDoc = iframe.contentDocument;
+  const iframeBody = iframeDoc.body;
+  const adBannerRoot =
```
Comment (aghassemi) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: If this has to be a direct child, use `childElementByTag` from `dom.js`

File: extensions/amp-lightbox/0.1/amp-lightbox.js [hunk 2]
```diff
@@ -159,6 +240,48 @@ class AmpLightbox extends AMP.BaseElement {
     this.active_ = true;
   }
 
+  /** @private */
+  maybeEnterFrameFullOverlayMode_() {
+    if (this.isInMainDocument_()) {
+      return;
+    }
+
+    if (!isExperimentOn(this.getAmpDoc().win, A4A_PROTOTYPE_EXPERIMENT)) {
```
Comment (aghassemi) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: move this up before the isInMainDocument to make the experiment completely side-effect free.

File: extensions/amp-lightbox/0.1/amp-lightbox.js [hunk 3]
```diff
@@ -159,6 +240,48 @@ class AmpLightbox extends AMP.BaseElement {
...
+    enterFrameFullOverlayMode(this.getIframe_(), this.getAmpDoc().win);
+  }
+
+  /** @private */
+  maybeLeaveFrameFullOverlayMode_() {
+    if (this.isInMainDocument_()) {
+      return;
+    }
+
+    if (!isExperimentOn(this.getAmpDoc().win, A4A_PROTOTYPE_EXPERIMENT)) {
```
Comment (aghassemi) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: ditto

File: examples/ad-lightbox.amp.html [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+</head>
+<body>
+  <!--
+    THIS EXAMPLE IS INVALID AMP.
+
+    It corresponds to an in-progress prototype. It will not validate or function
+    properly under normal circumstances.
+  -->
+
+  <amp-ad-banner layout="fill">
```
Comment (alanorozco) [inline examples/ad-lightbox.amp.html]: `amp-ad-banner` is unimplemented, so it's not needed. The document is not supposed to validate :)

File: examples/ad-lightbox.amp.html [hunk 5]
```diff
@@ -0,0 +1,103 @@
+<!doctype html>
```
Comment (alanorozco) [inline examples/ad-lightbox.amp.html]: Done.

File: examples/ad-lightbox.amp.html [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+<html ⚡>
+<head>
+    <meta charset="utf-8">
+    <title>AMP #0</title>
```
Comment (alanorozco) [inline examples/ad-lightbox.amp.html]: Done.

File: extensions/amp-lightbox/0.1/amp-lightbox.js (hunk 2 above)
Comment (alanorozco) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: Done.

File: extensions/amp-lightbox/0.1/amp-lightbox.js (hunk 3 above)
Comment (alanorozco) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: And done.

**Pull request merged**